doc/source/whatsnew/v1.2.0.rst

```diff
@@ -144,6 +144,8 @@ Deprecations
 ~~~~~~~~~~~~
 - Deprecated parameter ``inplace`` in :meth:`MultiIndex.set_codes` and :meth:`MultiIndex.set_levels` (:issue:`35626`)
 - Deprecated parameter ``dtype`` in :~meth:`Index.copy` on method all index classes. Use the :meth:`Index.astype` method instead for changing dtype(:issue:`35853`)
+- :func:`read_excel` "xlrd" engine is deprecated. The recommended engine is "openpyxl" for "xlsx" and "xlsm" files, because "xlrd" is no longer maintained (:issue:`28547`).
```

Comment:
thanks for updating. it's the ``read_excel` "xlrd" engine is deprecated` bit that I wanted removed

IIUC the xlrd is not deprecated. it's only that that default engine used will change.

Reply:
Is it deprecated for use with xlsx files? IOW in the future, will only openpyxl be supported for xlsx?  Sounds reasonable to me.

Reply:
hmm, that's not my understanding of the discussion in #28547

from https://github.com/pandas-dev/pandas/issues/28547#issue-496123767

> Considering that I think we need to deprecate using xlrd in favor of openpyxl. We might not necessarily need to remove the former and it does offer some functionality the latter doesn't (namely reading .xls files) but should at the very least start moving towards the latter

Indeed, the discussion has not explicitly mentioned disallowing `xlrd` for formats that `openpyxl` supports. but if the xlrd engine is not removed, we should decide now whether we would restrict it's use.

Reply:
My read of this is to only keep xlrd for xls where it is required, and to deprecate where it is not.  In the long run, if xlrd breaks and no one takes over its maintenance, then we will either have to vendor xlrd or remove support for xls. In either case minimizing use of xlrd seems like a good idea to me.

Reply:
@WillAyd is very specific regarding this issue. He states that we should warn first, then change the default (and maybe even remove the xlrd engine for xlsx and xlsm files altogether).
But not at this point, only a deprecation warning is asked, to notify users that this engine is no longer the preferred engine.

Reply:
Just to clarify we should only be changing the *default* reader to openpyxl. I think it's fine to keep xlrd around as a YMMV situation

Reply:
Hmm, I just reverted the changes that made openpyxl the default. I am now very confused. I thought that this change was just about warning people about pending deprecation of the xlrd reader, not to switch them over already.

Reply:
See your comments of July 1 and 3.

Reply:
I see where the wording is confusing, but yes we only warn now and change in the future. We always manage user-facing changes that way